bbc/simorgh
# NEWSWORLDSERVICE-1861 no-js for social embeds

Resolves JIRA [NEWSWORLDSERVICE-1861](https://jira.dev.bbc.co.uk/browse/NEWSWORLDSERVICE-1861)

Overall changes
======
Removes Lazy loading for social embeds and implements no-script fallback, re-using Notice created for social embeds.

Before:
![Screenshot 2024-02-07 at 13 11 26](https://github.com/bbc/simorgh/assets/4706195/84f01e98-d0c6-452d-9e9f-4777607259f2)

After:
![Screenshot 2024-02-07 at 13 11 37](https://github.com/bbc/simorgh/assets/4706195/a71ce2a8-7ec0-4fc8-86d1-7440cbed0d7a)


Code changes
======

- Adds `no-js` class script logic for NextJS app
- Adds a `<noscript>` to display already created notice for embed availability 

Testing
======
Needs regression test for pages with social media embeds
1. Run `yarn dev` in either ws-nextjs-app for Live pages or root directory for the rest
2. Visit http://localhost:7081/pidgin/live/c7p765ynk9qt?renderer_env=test, http://localhost:7080/mundo/articles/c20myn5q195o and http://localhost:7080/mundo/23263889 (Test equivalent https://www.test.bbc.com/pidgin/live/c7p765ynk9qt, https://www.test.bbc.com/mundo/articles/c20myn5q195o  and https://www.test.bbc.com/mundo/23263889)
3. Use a tool to disable JS such as [Web Developer](https://chromewebstore.google.com/detail/web-developer/bfbameneiokkgbdmiekhjnmfkcnldhhm)
4. See that Notice shows

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
